src/java.net.http/share/classes/jdk/internal/net/http/PullPublisher.java

```diff
@@ -102,7 +102,7 @@ protected void run() {
                     return;
                 }
 
-                while (demand.tryDecrement() && !cancelled) {
+                while (demand.tryDecrement() && !cancelled && error == null) {
```

Comment:
The changes to `PullPublisher` seem like a general bug fix unrelated to `concat`. Is that true? Does it make sense to separate them out, or are they only applicable to a very narrow boundary scenario?